standard-libraries/iterator.3.mr

```diff
@@ -0,0 +1,15 @@
+## Generic iterator dynamic interface
+
+
+class Iterator{Generic}
+  ## Return whether current item exist
+  func dynamic has-item():(copy Bool exist)
+    raise
+  
+  ## Get current item, fail if not exist
+  func dynamic get-item():(user Generic item)
+    raise
```

Comment:
This is temporary, error-handling syntax is not decided yet (#5).
This is just converted to `return ERR;` in C.